### PR TITLE
Refactor expenditure and budget attributes

### PR DESCRIPTION
**Pull request by sogggy**

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -89,13 +88,13 @@ public static Price parsePrice(String price) throws ParseException {
      *
      * @throws ParseException if the given {@code name} is invalid.
      */
-    public static BudgetName parseBudgetName(String budgetName) throws ParseException {
+    public static Name parseBudgetName(String budgetName) throws ParseException {
         requireNonNull(budgetName);
         String trimmedName = budgetName.trim();
-        if (!BudgetName.isValidBudgetName(trimmedName)) {
-            throw new ParseException(Name.MESSAGE_CONSTRAINTS);
+        if (!Name.isValidBudgetName(trimmedName)) {
```
Comment (wenhaogoh) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java]: Perhaps it is more appropriate if the method "isValidBudgetName" is changed to "isValid"?

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 2]
```diff
@@ -89,13 +88,13 @@ public static Price parsePrice(String price) throws ParseException {
      *
      * @throws ParseException if the given {@code name} is invalid.
      */
-    public static BudgetName parseBudgetName(String budgetName) throws ParseException {
+    public static Name parseBudgetName(String budgetName) throws ParseException {
```
Comment (wenhaogoh) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java]: Perhaps it is more appropriate if the argument "budgetName" is changed to "name"?

**Review by wenhaogoh — APPROVED** [review]
LGTM 👍

**Review by EkChinHui — APPROVED** [review]
LGTM 👍

**Pull request merged**